### PR TITLE
added api

### PR DESCRIPTION
**Pull request by mmarganski**

### CODE REVIEW

File: ztm-front/src/components/AddPlaceModal.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                <Overlay onClick={onClose}/>
+                <Wrapper>
+                    <Input
+                        placeholder={'Name of your place'}
+                        value={placeName}
+                        onChange={event => setPlaceName(event.target.value)}
+                        onKeyDown={event => event.key === 'Enter' ? submitInput() : null}
+                    />
+                    <Input
+                        placeholder={'radius of influence'}
```
Comment (jpudysz) [inline ztm-front/src/components/AddPlaceModal.tsx]: banned-syntax

File: ztm-api/src/app.gateway.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        }
+
+        const response = await this.makeRequest(`${process.env.ZTM_URL}${this.currentBuses.join(',')}`)
+        const jsonResponse = convert.xml2json(response.data, {compact: false, spaces: 4})
+        const arrayData = Object.values(
+            JSON.parse(jsonResponse)
+                ?.elements
+                ?.[0]
+                ?.elements
+        ).map((element: outerElement) => element
```
Comment (gtarnopolski) [inline ztm-api/src/app.gateway.ts]: indentation

File: ztm-front/src/components/ApiConnector.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import { useStore } from 'outstated'
+import { LatLng } from '../types'
+import {
+    useActiveBussesStore,
+    useBusListStore,
+    useBusPositionsStore
+} from '../stores'
+
+export const ApiConnector: React.FunctionComponent = ({ children }) => {
+    const [socketConnection, setSocketConnection] = useState<Socket<DefaultEventsMap, DefaultEventsMap>>()
```
Comment (gtarnopolski) [inline ztm-front/src/components/ApiConnector.tsx]: why you used generic type here not just Socket

File: ztm-front/src/components/ApiConnector.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+import { DefaultEventsMap } from 'socket.io-client/build/typed-events'
+import { useStore } from 'outstated'
+import { LatLng } from '../types'
+import {
+    useActiveBussesStore,
+    useBusListStore,
+    useBusPositionsStore
+} from '../stores'
+
+export const ApiConnector: React.FunctionComponent = ({ children }) => {
```
Comment (gtarnopolski) [inline ztm-front/src/components/ApiConnector.tsx]: socket should be in separate store

File: ztm-front/src/components/ApiConnector.tsx (hunk 3 above)
Comment (jpudysz) [inline ztm-front/src/components/ApiConnector.tsx]: I would use ref here

File: ztm-front/src/components/ApiConnector.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+    socketConnection?.on('busesList', (busesList: Array<Array<string>>) => {
+        setBusList(Object.fromEntries(busesList))
+    })
+
+    useEffect(() => {
+        setSocketConnection(io('http://localhost:3003/', { transports: ['websocket'] }))
+    }, [])
+
+    useEffect(() => {
+        socketConnection?.emit('getBusesList')
```
Comment (gtarnopolski) [inline ztm-front/src/components/ApiConnector.tsx]: emits, should be actions returned from store

**Review by jpudysz — CHANGES_REQUESTED** [review]

File: ztm-front/src/components/Main.tsx [hunk 6]
```diff
@@ -1,19 +1,25 @@
...
-    const [currentSelect, setSelect] = useState<ButtonsList>(ButtonsList.TrackBus)
+    const [currentSelect, setSelect] = useState<TabsList>(TabsList.TrackBus)
 
     return(
         <MainWrapper>
             <Sidebar
-                buttons={Object.values(ButtonsList)}
-                onSidebarSelect={(tabName: ButtonsList) => setSelect(tabName)}
+                buttons={Object.values(TabsList)}
+                onSidebarSelect={(tabName: TabsList) => setSelect(tabName)}
```
Comment (gtarnopolski) [inline ztm-front/src/components/Main.tsx]: proxy

File: ztm-front/src/stores/useActiveBussesStore.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { useState } from 'react'
+
+export const useActiveBussesStore = () => {
+    const [activeBuses, setActiveBuses] = useState<Array<string>>([])
+
+    const toggleBus = (newBusId: string) => {
```
Comment (gtarnopolski) [inline ztm-front/src/stores/useActiveBussesStore.ts]: id in definition should be number

File: ztm-front/src/stores/useBusListStore.ts [hunk 8]
```diff
@@ -0,0 +1,10 @@
+import { useState } from 'react'
+
+export const useBusListStore = () => {
+    const [busList, setBusList] = useState<Record<string, string>>({})
```
Comment (gtarnopolski) [inline ztm-front/src/stores/useBusListStore.ts]: Store in it's definition should be a container -> in your case it is created on each `useState`

there should be `useBus` and it should contain everything related to bus, and so on with other examples,
in your scenario in bigger project you will have 100+ stores, when there can be 10 at most

**Review by gtarnopolski — CHANGES_REQUESTED** [review]

**Review by jpudysz — APPROVED** [review]

File: ztm-front/src/components/ApiConnector.tsx [hunk 9]
```diff
@@ -1,42 +1,53 @@
...
-        socketConnection?.emit('getBusesList')
-    }, [socketConnection])
+        setActiveBusses(activeBuses)
+    }, [activeBuses])
 
     useEffect(() => {
-        socketConnection?.emit('setActiveBusses', activeBuses)
-    }, [activeBuses])
+        const matchingPlaces = places.filter(place => activePlaces.includes(place.name))
+        busPositions.forEach(position => {
```
Comment (gtarnopolski) [inline ztm-front/src/components/ApiConnector.tsx:41]: enter

**Review by gtarnopolski — APPROVED** [review]

**Pull request merged**